Shared/Supporting Files/Views/CategoriesView.swift

```diff
@@ -149,3 +171,11 @@ private extension CategoriesView {
         }
     }
 }
+
+extension EnvironmentValues {
+    /// The action to run when a sample is done tearing down.
+    ///
+    /// - Note: Failure to call this in a ``Sample/hasTeardown`` sample will
+    /// block other teardown samples from appearing.
+    @Entry var onTearDownCompleted: () -> Void = {}
```

Comment:
`onTeardown` is a great name for a modifier, but an odd name for an environment value. Since this is an action, a better name would be something like `finishTeardown`.